bundle-workflow/infra/lib/distribution-build-stack.ts

```diff
@@ -0,0 +1,100 @@
+import { CloudFrontAllowedMethods, CloudFrontWebDistribution, OriginAccessIdentity } from '@aws-cdk/aws-cloudfront';
+import {
+  ArnPrincipal, CanonicalUserPrincipal, IRole, PolicyStatement, Role,
+} from '@aws-cdk/aws-iam';
+import { Bucket } from '@aws-cdk/aws-s3';
+import {
+  Arn, CfnOutput, CfnParameter, Construct, Stack, StackProps,
+} from '@aws-cdk/core';
+
+export interface DistributionBuildStackProps extends StackProps {
+  name: string,
+  bucketName?: string,
+  buildRoleName?: string,
+  bundleRoleName?: string,
+  testRoleName?: string,
+}
+
+export class DistributionBuildStack extends Stack {
```

Comment:
If you are looking for the 'meat' of this change as it were - this is it.  Everything else is support for the CDK build/deployment process.